projects/demo/src/pages/documentation/angular/angular.template.html

```diff
@@ -56,50 +52,62 @@ <h2>Basic directive approach</h2>
+                to find input element for you, if you do not have a direct
+                access to it.
             </p>
 
-            <p>
-                Let's imagine that you have a fancy-looking input component that
-                satisfies your design system:
-            </p>
+            <tui-notification>
+                By default
+                <strong>maskito</strong>
+                will try to find input/textarea by querying its host:
+                <code>host.querySelector('input,textarea')</code>
+                so that might be sufficient. Use custom predicate if you need
+                custom logic.
+            </tui-notification>
 
             <tui-doc-code
                 filename="custom-input.component.ts"
                 [code]="customInputExample"
             ></tui-doc-code>
-
-            <p>
-                This solution causes props drilling. You should pass
-                <code>maskitoOptions</code>
-                through all components (which don't even requires these options)
-                until we reached native input. This example is very simple, but
-                in real life you can have a lot of nested components inside
-                custom input.
-            </p>
-
-            <p>
-                Use Dependency Injection approach for the such case. Just put
-                empty
-                <code>maskito</code>
-                -attribute on native input inside your custom input component.
-            </p>
-
-            <tui-doc-code
-                filename="custom-input.component.ts"
-                [code]="customInputDiApproachExample"
-            ></tui-doc-code>
-
-            <p>
-                And now users of your UI Kit can put any options they want to
-                your custom input:
-                <tui-doc-code [code]="diApproachInAction"></tui-doc-code>
-            </p>
         </section>
+
+        <tui-doc-example
+            heading="Custom input"
+            description="See querying nested input in action"
+            [content]="nestedInputExample"
+        >
+            <tui-notification class="example tui-space_bottom-3">
+                Default behavior is enough for
+                <a
+                    tuiLink
+                    href="https://github.com/Tinkoff/taiga-ui"
+                >
+                    Taiga UI
+                </a>
+                inputs
+            </tui-notification>
+
+            <nested-doc-example-1
+                #example
+                class="example"
+            ></nested-doc-example-1>
+
+            <tui-notification class="example tui-space_vertical-3">
+                Custom predicate is required if target input is not the first on
+                in the DOM
+            </tui-notification>
+
+            <nested-doc-example-2
+                class="example"
+                [maskito]="example.nameMask"
+                [maskitoElement]="example.predicate"
```

Comment:
Let't move this logic inside
`projects/demo/src/pages/documentation/angular/examples/2-nested/template.html`.

**Proposed content:**
```html
<div [maskito]="..." [maskitoElement]="...">
  <tui-checkbox-labeled
      class="tui-space_bottom-3"
      [(ngModel)]="show"
  >
      Add card holder name
  </tui-checkbox-labeled>
  <tui-input
      [disabled]="!show"
      [(ngModel)]="value"
  >
      Name on the card
  </tui-input>
</div>
```

it is more obvious for the reader of the documentation.

Reply:
The idea was that this template file is a particular component code where both checkbox and input are wrapped inside some `<app-card-holder></app-card-holder>` component where you do not have access to the `tui-input`. Do you think we need a better example showcasing this?

Reply:
When i read firstly this [page](https://maskito--pr164-predicate-4i3qil38.web.app/angular) (and didn't review code yet), i didn't understand where was there maskito inside `Custom` tab.

And it took some time to understand that maskito somewhere outside.
We can place two component inside `Custom`-tab: AppComponent and CustomInputComponent. What do you think ?